File: frontend/types/movie.ts
export interface Movie {
  id: string;
  name: string;
  description: string;
  genre?: string;
  rating?: number;
  imageUrl?: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateMovieDto {
  name: string;
  description: string;
  genre?: string;
  rating?: number;
  image?: File;
  imageUrl?: string;
}

export interface UpdateMovieDto {
  name: string;
  description: string;
  genre?: string;
  rating?: number;
  image?: File;
  imageUrl?: string;
}

File: Backend/Models/Movie.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace Backend.Models;

public class Movie
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string? Id { get; set; }

    [BsonElement("name")]
    [BsonRequired]
    public string Name { get; set; } = string.Empty;

    [BsonElement("description")]
    [BsonRequired]
    public string Description { get; set; } = string.Empty;

    [BsonElement("genre")]
    public string? Genre { get; set; }

    [BsonElement("rating")]
    public int? Rating { get; set; }

    [BsonElement("imageUrl")]
    public string? ImageUrl { get; set; }

    [BsonElement("createdAt")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [BsonElement("updatedAt")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}
